packages/debug/src/node/debug-adapter.ts

```diff
@@ -60,9 +63,19 @@ export class LaunchBasedDebugAdapterFactory implements DebugAdapterFactory {
         };
     }
 
-    private spawnProcess(executable: DebugAdapterExecutable): RawProcess {
-        const { command, args } = executable;
-        return this.processFactory({ command, args, options: { stdio: ['pipe', 'pipe', 2] } });
+    private childProcess(executable: DebugAdapterExecutable): RawProcess {
+        const isForkOptions = (forkOptions: RawForkOptions | any): forkOptions is RawForkOptions =>
+            !!forkOptions && !!forkOptions.modulePath;
+
+        const processOptions: RawProcessOptions | RawForkOptions = executable;
+        const options = { stdio: ['pipe', 'pipe', 2] };
+
+        if (isForkOptions(processOptions)) {
+            options.stdio.push('ipc');
+        }
+
+        processOptions.options = options;
```

Comment:
looks like we mutate `executable` here, we should create a new copy

Reply:
ack, have updated the code to shallow-copy executable.